modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/RuleEditor/RuleEditor.es.js

```diff
@@ -164,25 +164,9 @@ class RuleEditor extends Component {
 
 		deletedFields: Config.arrayOf(Config.string()).value([]),
 
-		fieldOptions: Config.arrayOf(
-			Config.shapeOf(
-				{
-					dataType: Config.string(),
-					name: Config.string(),
-					options: Config.arrayOf(
-						Config.shapeOf(
-							{
-								label: Config.string(),
-								name: Config.string(),
-								value: Config.string()
-							}
-						)
-					),
-					type: Config.string(),
-					value: Config.string()
-				}
-			)
-		).internal().valueFn('_fieldOptionsValueFn'),
+		fieldOptions: Config.arrayOf(fieldOptionStructure).internal().valueFn('_fieldOptionsValueFn'),
+
+		fieldOptionsWithRestrictions: Config.arrayOf(fieldOptionStructure).internal().valueFn('_fieldOptionsWithRestrictionsValueFn'),
```

Comment:
@alinedoleron I don't understand why we have both fieldOptions and fieldOptionsWithRestrictions. The restriction could be just the way fieldOptions work.